libnm-core/nm-utils.c

```diff
@@ -2455,21 +2482,45 @@ nm_utils_tc_qdisc_from_str (const char *str, GError **error)
 	gs_free char *kind = NULL;
 	gs_free char *rest = NULL;
 	NMTCQdisc *qdisc = NULL;
-	gs_unref_hashtable GHashTable *ht = NULL;
+	gs_unref_hashtable GHashTable *options = NULL;
+	GHashTableIter iter;
+	gpointer key, value;
+	int i;
 
 	nm_assert (str);
 	nm_assert (!error || !*error);
 
-	ht = nm_utils_parse_variant_attributes (str,
+	options = nm_utils_parse_variant_attributes (str,
 	                                        ' ', ' ', FALSE,
 	                                        tc_object_attribute_spec,
 	                                        error);
-	if (!ht)
+	if (!options)
 		return NULL;
+	g_clear_pointer (&options, g_hash_table_unref);
```

Comment:
`nm_clear_pointer()` instead of `g_clear_pointer()`. It does not cast the pointer to void, and you'd get a compiler error when the function's signature does not match.

Reply:
This just throws away `@options`. That doesn't look right, shouldn't it consider what was parsed there?